web/src/routes/(user)/albums/[albumId=id]/[[photos=photos]]/[[assetId=id]]/+page.svelte

```diff
@@ -399,6 +400,13 @@
       handleError(error, 'Unable to update album cover');
     }
   };
+
+  const checkAlbumAndGoBack = async () => {
+    if (album.assetCount === 0) {
```

Comment:
This check changes the behavior slightly. Previously, an empty album name was required before the album gets deleted. Is that an intentional change?

Reply:
No, that was by mistake. Thanks for pointing it out.

Reply:
The album is still deleted even when setting a title., it looks like the property is not updated before the page is closed and the album reloaded.

I have to look into it.

Reply:
You can use bind to make sure the album name is updated:

```diff
- <AlbumTitle id={album.id} albumName={album.albumName} {isOwned} />
+ <AlbumTitle id={album.id} bind:albumName={album.albumName} {isOwned} />
```

Maybe using [beforeNavigate](https://kit.svelte.dev/docs/modules#$app-navigation-beforenavigate) would be a more robust approach, because there are several ways to navigate away from this page including the browser back button, pressing escape and clicking the back arrow.

Reply:
I had another look into it. The reason is, that after a full page reload, the `backUrl` is set to `ALBUMS` instead of `SHARING` (the `from` route gets lost due to the reload). So when clicking the "close" button, the user is redirected to "Albums" instead of "Sharing" (this is also the case on `main`).

The `load` method in "albums" is triggered/finished before the `delete` call in `onNavigate` which leads to the aforementioned error (`getAllAlbums` loads a soon to be deleted album). It seems like SvelteKit executes the `load` function before `onNavigate` completes (this was surprising to me but, as mentioned, I'm not an expert in Svelte).

If I'm not mistaken, the only way to solve the issue is to create the album in the "createAlbum" page itself, but this is bigger change. Not sure if it's worth the effort...

Another option would be to go back to the previous "onClose" solution with the drawback that it only covers the close by button case.

Any other ideas?

Reply:
The back url defaulting to /albums is expected, because the state isn't persisted.

It's annoying that load functions run first. I did come up with a solution that seems to work by first cancelling navigation and then calling `goto`, but it's nothing pretty:

```ts
let albumDeleted = false;

beforeNavigate((navigation) => handlePromiseError(handleBeforeNavigate(navigation)));

const handleBeforeNavigate = async (navigation: BeforeNavigate) => {
  const url = navigation.to?.url;
  if (!url || albumDeleted) {
    return;
  }

  if (album.assetCount === 0 && !album.albumName) {
    navigation.cancel();
    albumDeleted = true;

    try {
      await deleteAlbum(album);
    } finally {
      await goto(url);
    }
  }
};
```

Another solution would be to treat `400 Bad Request` as a success when deleting empty albums on the page `/albums`. Typically delete calls for REST APIs are idempotent and this wouldn't be an issue, but ours isn't.

Reply:
I also thought about loading the albums outside of the `load` function, but this feels hacky as well. In addition, if I'm not mistaken, the `load` function runs server side so it would probably have performance implications.

If I had to choose, I would go with ignoring the error on the album page, as it may cover other such cases (although I can't think of any). Since I don't like either solution (no offense), I'd go with the one that's easiest to implement. 

Of course, it's your final decision.

Reply:
Ignoring 400 errors seems fine to me